src/useVisibilityObserver/useVisibilityObserver.ts

```diff
@@ -0,0 +1,33 @@
+import { onMounted, ref, Ref } from 'vue'
+import { useIntersectionObserver, UseIntersectionObserverOptions } from '@/useIntersectionObserver'
+
+export type UseVisibilityObserverResponse = {
+  visible: Ref<boolean>,
+  disconnect: () => void,
+}
+
+export type UseVisibilityObserverOptions = UseIntersectionObserverOptions & { disconnectWhenVisible: boolean }
+
+export function useVisibilityObserver(element: Ref<HTMLElement | undefined>, options: UseVisibilityObserverOptions = { disconnectWhenVisible: false }): UseVisibilityObserverResponse {
+  const visible = ref(false)
+
+  function intersect(entries: IntersectionObserverEntry[]): void {
+    entries.forEach(entry => {
+      visible.value = entry.isIntersecting
+
+      if (options.disconnectWhenVisible && entry.isIntersecting) {
+        disconnect()
+      }
+    })
+  }
+
+  const { root, rootMargin, threshold } = options
+
+  const { observe, disconnect } = useIntersectionObserver(intersect, { root, rootMargin, threshold })
```

Comment:
Slight change to how options are used. This will prevent this composition from needing to be updated if the options for `useIntersectionObserver` change. 

```suggestion
  const visible = ref(false)
  const { disconnectWhenVisible, ...intersectionObserverOptions } = options

  function intersect(entries: IntersectionObserverEntry[]): void {
    entries.forEach(entry => {
      visible.value = entry.isIntersecting

      if (disconnectWhenVisible && entry.isIntersecting) {
        disconnect()
      }
    })
  }

  const { observe, disconnect } = useIntersectionObserver(intersect, intersectionObserverOptions)
```